### DISCUSSION TITLE
Invalid character in Remote JSON authorizer for payload.

### DISCUSSION THREAD

**Discussion by rezakazemi928** [ory/oathkeeper#943] [Q&A] [answered]
Hi,
I have an unclear issue and I would appreciate for little help. I have set up a remote JSON handler as an authorizer for my configuration and I deploy it successfully on a server. 
here is my accessrules.yml

```
 id: some ID
  upstream:
    preserve_host: true
    url: https://httpbin.org
  match:
    url: https://match_url:4456/<**>
    methods:
    - GET
    - POST
    - PUT
    - DELETE
    - PATCH
  authenticators:
  - handler: cookie_session
  authorizer: 
    handler: remote_json
    config:
      remote: http://api/accessrules/authorization
      payload: |
        {
            "subject": "{{ print .Subject }}",
            "url": "{{ print .MatchContext.URL }}",
            "header": "{{ print .MatchContext.Header }}"
        }
  mutators:
  - handler: noop 
  errors:
  - handler: json

```

I had tested my performance on firefox and everything was just fine.
when I used chrome browser I ran to an issue for this error on the deployment:

`"error":{"message":"payload is not a JSON text: invalid character 'N' after object key:value pair"},`

I spend a long time debugging the problem and I find out the only reason I get this message is I had added the header key to the payload.

I do not know How to solve the problem.

**Comment by gen1us2k** [comment, marked as answer]
Hello, @rezakazemi928 . What matching strategy do you use? Is it regexp or glob? I assume that you use the glob matching strategy and you can have this issue with glob because oathkeeper does not support glob to provide MatchContext at the moment. Your fields `url` and `header` remain empty when proxying to the upstream. `Regexp` matching strategy is fully supported and consider using this strategy to glob
